project-management/src/main/java/life/qbic/projectmanagement/application/measurement/MeasurementService.java

```diff
@@ -674,6 +611,124 @@ public CompletableFuture<List<Result<MeasurementId, ErrorCode>>> updateAll(
     return CompletableFuture.completedFuture(results);
   }
 
+  /**
+   * Bulk update of a list of proteomics measurements.
+   * <p>
+   * This method takes care of two types of measurements:
+   *
+   * <ul>
+   *   <li>single sample measurement</li>
+   *   <li>pooled sample measurement</li>
+   * </ul>
+   * <p>
+   * In the case of pooled sample measurement updates, the method groups
+   * the metadata entries by measurement ID since the pool has already been registered and
+   * cannot be reassigned.
+   * <p>
+   * In the current state of implementation, if a user wants to update the pool label, they have to
+   * delete the measurement and register it properly again.
+   *
+   * @param metadata  the proteomics metadata to update
+   * @param projectId the project the measurement belongs to
+   * @return a list of {@link Result} objects
+   * @since 1.0.0
+   */
+  private List<Result<MeasurementId, ErrorCode>> updateAllPxp(
+      List<ProteomicsMeasurementMetadata> metadata, ProjectId projectId) {
+
+    if (measurementCodeMissing(metadata)) {
+      throw new MeasurementRegistrationException(ErrorCode.MISSING_MEASUREMENT_ID);
+    }
+    if (!allMeasurementCodesExist(
+        metadata.stream().map(ProteomicsMeasurementMetadata::measurementId).toList())) {
+      throw new MeasurementRegistrationException(ErrorCode.UNKNOWN_MEASUREMENT);
+    }
+
+    var singleSampleMeasurements = metadata.stream()
+        .filter(measurement -> measurement.assignedSamplePoolGroup().isEmpty()).toList();
+    var pooledMeasurements = metadata.stream()
```

Comment:
what happens with measurements where a pool group is provided but no other samples are included in the given pool group? Does it matter that you categorize here?
It looks to me as if some code is duplicated here.

Reply:
Then the pool is of size one, which is up to the user to define if it is meaningful or not.

Reply:
Yes but the code does not need to differentiate. You have a lot of duplicate logic in the two if branches.

Reply:
Apologies, I dont understand the issue with the referenced code line